### DISCUSSION TITLE
`git fetch` results in a lot of new remote branches getting processed

### DISCUSSION THREAD

**Discussion by pink** [arxanas/git-branchless#879] [Q&A] [answered]
Want to preface this with a big thanks for creating this project! It's very slick and adds some much missed functionality back to git.

One thing I've noticed is that when I run `git fetch`, I'll get a bunch (sometimes hundreds) of new remote branches added to git-branchless. It's a wall of text in my terminal so I don't think it's worth posting here, but each line is basically:
```
branchless: processing 1 update: remote branch origin/<branch_name>
 * [new branch]              <branch_name>                              -> origin/<branch_name>
```

Almost all of these updates are remote branches pushed by other contributors to the same repo. My questions:
* Is there a reason why git-branchless needs to index these?
* Is it possible to suppress this text?
* Are there any major downsides to only fetching the `master`/`main` branch?

**Comment by arxanas** [comment, marked as answer]
git-branchless prints these as part of the `reference-transaction` hook. Ideally, these branches would all be batched into the same transaction, so that only one (very long) line would be printed, but upstream Git hasn't implemented that.

git-branchless used to work with remote branches, but it no longer does, so I guess we could ignore updates to remote branches. If you want to make that change, you'll have to do it in the `reference-transaction` hook:

https://github.com/arxanas/git-branchless/blob/de7bf6911829f2e3a85bdbe04640e327fdedee98/git-branchless-hook/src/lib.rs#LL474C41-L474C41

There's currently no way to suppress the printing of the message itself.

Fetching only the main branch is fine (that's what I usually do), with the obvious caveat that then you won't have immediate access to your colleagues' branches. You can fetch those branches on-demand later with `git fetch origin <branch>`.

**Reply by pink** [reply]
Ah got it, thanks for the explanation! Is there anywhere I can read up on how git-branchless used to interact with remote branches? I think just fetching the main branch by default should work for now, but I can take a stab at updating `reference-transaction` as well.

**Reply by arxanas** [reply]
Specifically, we supported having the main branch be a remote branch. It also happened that, because we tracked updates to remote branches, you could undo the result of a local `git fetch` — but that turned out to be more confusing than useful in practice, because the local and remote repository would disagree on the positions of refs after the undo and commands like `git push --force-with-lease` would fail.

You can find the announcement post on dropping remote branch support here: https://github.com/arxanas/git-branchless/discussions/595

**Reply by pink** [reply]
Put up a PR for this change - let me know if this is what you had in mind

https://github.com/arxanas/git-branchless/pull/913
